sphinx_ext_substitution/substitution.py

```diff
@@ -66,10 +66,12 @@ def sub_role(name, rawtext, text, lineno, inliner,
         original = original.replace('\x00`', '`')
 
     # Find the replacement value, don't use it for anything yet.
-    if id_ in subs:
-        replacement = subs[id_]
-    else:
-        replacement = None
+    replacement = None
+    for potential_id, value in subs.items():
```

Comment:
I'm a bit more cautious here: what happens when the number of substitutions gets very long?  I wouldn't imagine that for my use cases either, but it could happen.

If it helps you a lot, I guess I wouldn't mind doing it, but I'm a bit worried about committing to support this forever since I don't know how much this extension will be used in the future.  Also things like short IDs being unintentionally sub-strings of other things.  Actually, I guess this is more common in my imagined use cases...

Idea: make this a configurable option, default off, call it "in testing", and see what happens?